test/e2e/suite/cases/audit/audit.go

```diff
@@ -0,0 +1,269 @@
+	})
+	Expect(err).NotTo(HaveOccurred())
+	if len(pods.Items) != 1 {
+		Expect(fmt.Errorf("expected single kube-oidc-proxy pod running, got=%d", len(pods.Items))).NotTo(HaveOccurred())
+	}
+
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "kube-oidc-proxy-e2e")
+	Expect(err).NotTo(HaveOccurred())
+
+	defer func() {
+		err := os.RemoveAll(tmpDir)
+		Expect(err).NotTo(HaveOccurred())
+	}()
+
+	logFile := filepath.Join(tmpDir, "log.txt")
+
+	err = f.Helper().Kubectl(f.Namespace.Name).Run(
+		"cp", fmt.Sprintf("%s:audit-log", pods.Items[0].Name), logFile)
+	Expect(err).NotTo(HaveOccurred())
+
+	logs, err := ioutil.ReadFile(logFile)
+	Expect(err).NotTo(HaveOccurred())
+
+	scanner := bufio.NewScanner(bytes.NewReader(logs))
+
+	expAuditEvents := []auditv1.Event{
+		auditv1.Event{
+			Level:      auditv1.LevelRequestResponse,
+			Stage:      auditv1.StageRequestReceived,
+			RequestURI: "/api/v1/namespaces/kube-system/pods",
+			Verb:       "get",
+			User: authnv1.UserInfo{
+				Username: "user@example.com",
+				Groups:   []string{"group-1", "group-2"},
+			},
+		},
+		auditv1.Event{
+			Level:      auditv1.LevelRequestResponse,
+			Stage:      auditv1.StageResponseComplete,
+			RequestURI: "/api/v1/namespaces/kube-system/pods",
+			Verb:       "get",
+			User: authnv1.UserInfo{
+				Username: "user@example.com",
+				Groups:   []string{"group-1", "group-2"},
+			},
+			ResponseStatus: &metav1.Status{
+				Code: 403,
+			},
+		},
+		auditv1.Event{
+			Level:      auditv1.LevelRequestResponse,
+			Stage:      auditv1.StageResponseStarted,
+			RequestURI: "/api/v1/namespaces/kube-system/pods",
+			Verb:       "get",
+			ResponseStatus: &metav1.Status{
+				Code:    401,
+				Message: "Authentication failed, attempted: bearer",
+			},
+		},
+	}
+
+	By("Testing for expected audit logs")
+	var i int
+	for scanner.Scan() {
+		if i > len(expAuditEvents) {
+			Expect(fmt.Errorf("more proxy audit logs than expected, exp=%d got=%s", len(expAuditEvents), logs)).NotTo(HaveOccurred())
+		}
+
+		var auditEvent auditv1.Event
+		err = json.Unmarshal(scanner.Bytes(), &auditEvent)
+		Expect(err).NotTo(HaveOccurred())
+
+		gotAuditEvent := auditv1.Event{
+			Level:      auditEvent.Level,
+			Stage:      auditEvent.Stage,
+			RequestURI: auditEvent.RequestURI,
+			Verb:       auditEvent.Verb,
+			User: authnv1.UserInfo{
+				Username: auditEvent.User.Username,
+				Groups:   auditEvent.User.Groups,
+			},
+		}
+
+		if auditEvent.ResponseStatus != nil {
+			gotAuditEvent.ResponseStatus = &metav1.Status{
+				Code:    auditEvent.ResponseStatus.Code,
+				Message: auditEvent.ResponseStatus.Message,
+			}
+		}
+
+		if !reflect.DeepEqual(expAuditEvents[i], gotAuditEvent) {
+			Expect(fmt.Errorf("unexpected audit event\nexp=%v\ngot=%v", expAuditEvents[i], gotAuditEvent)).NotTo(HaveOccurred())
+		}
+
+		i++
+	}
+
+	if i != len(expAuditEvents) {
+		Expect(fmt.Errorf("less proxy audit logs then expected, exp=%d, got=%s", len(expAuditEvents), logs)).NotTo(HaveOccurred())
```

Comment:
nit: should be `len(logs)`

Reply:
I'm printing out the full raw log output here since it's better to have the actual json to inspect, rather than just the number of objects when debugging